### PR TITLE
JAMES-3521 Fix unstable integration tests in distributed server

### PR DESCRIPTION
**Pull request by LanKhuat**
Some integration tests running upon distributed server are failing because of fixed ID string, generating ID using default factory...

### CODE REVIEW

**Review by chibenwa — APPROVED** [review]

File: server/protocols/jmap-rfc-8621-integration-tests/distributed-jmap-rfc-8621-integration-tests/src/test/java/org/apache/james/jmap/rfc8621/distributed/DistributedMailboxSetMethodTest.java [hunk 1]
```diff
@@ -64,4 +67,14 @@ public MailboxId randomMailboxId() {
...
+
+    @Override
+    @Test
+    @Disabled("Distributed event bus is asynchronous, we cannot expect the newState to be returned immediately after Mailbox/set call")
+    public void newStateShouldBeUpToDate(GuiceJamesServer server) {}
+
+    @Override
+    @Test
+    @Disabled("Distributed event bus is asynchronous, we cannot expect the newState to be returned immediately after Mailbox/set call")
+    public void oldStateShouldIncludeSetChanges(GuiceJamesServer server) {}
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621-integration-tests/distributed-jmap-rfc-8621-integration-tests/src/test/java/org/apache/james/jmap/rfc8621/distributed/DistributedMailboxSetMethodTest.java:79]: Could not maybe rewrite it, by separating /set and /changes call? You get the old state from the /set response and put an await on your changes call?

File: server/protocols/jmap-rfc-8621-integration-tests/distributed-jmap-rfc-8621-integration-tests/src/test/java/org/apache/james/jmap/rfc8621/distributed/DistributedMailboxSetMethodTest.java [hunk 2]
```diff
@@ -64,4 +67,14 @@ public MailboxId randomMailboxId() {
     public String errorInvalidMailboxIdMessage(String value) {
         return String.format("%s is not a mailboxId: Invalid UUID string: %s", value, value);
     }
+
+    @Override
+    @Test
+    @Disabled("Distributed event bus is asynchronous, we cannot expect the newState to be returned immediately after Mailbox/set call")
+    public void newStateShouldBeUpToDate(GuiceJamesServer server) {}
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621-integration-tests/distributed-jmap-rfc-8621-integration-tests/src/test/java/org/apache/james/jmap/rfc8621/distributed/DistributedMailboxSetMethodTest.java:74]: Could not maybe rewrite it, by separating /set and /changes call? You get the new state from the /set response and put an await on your changes call?

File: server/protocols/jmap-rfc-8621-integration-tests/distributed-jmap-rfc-8621-integration-tests/src/test/java/org/apache/james/jmap/rfc8621/distributed/DistributedMailboxSetMethodTest.java (hunk 1 above)
Comment (LanKhuat) [reply server/protocols/jmap-rfc-8621-integration-tests/distributed-jmap-rfc-8621-integration-tests/src/test/java/org/apache/james/jmap/rfc8621/distributed/DistributedMailboxSetMethodTest.java:79]: Thing is at the point when the update completes, the new state might not been stored yet so we can't reliably extract it from the first call. 

If we use await on the state storing process then this test will be a duplication of a similar one from the MailboxChangesMethodContract.

File: server/protocols/jmap-rfc-8621-integration-tests/distributed-jmap-rfc-8621-integration-tests/src/test/java/org/apache/james/jmap/rfc8621/distributed/DistributedMailboxSetMethodTest.java (hunk 2 above)
Comment (LanKhuat) [reply server/protocols/jmap-rfc-8621-integration-tests/distributed-jmap-rfc-8621-integration-tests/src/test/java/org/apache/james/jmap/rfc8621/distributed/DistributedMailboxSetMethodTest.java:74]: idem

**Review by Arsnael — APPROVED** [review]

**Comment by LanKhuat** [comment]
`
org.awaitility.core.ConditionTimeoutException: Condition with lambda expression in org.apache.james.webadmin.integration.vault.DeletedMessageVaultIntegrationTest was not fulfilled within 2 minutes.
`
how do i restart the build?

**Comment by Arsnael** [comment]
Merged

**Pull request closed without merging**